static/js/tools/map/fetcher/denom_stat.ts

```diff
@@ -20,60 +20,83 @@
 
 import axios from "axios";
 import _ from "lodash";
-import { Dispatch } from "react";
+import { Dispatch, useContext, useEffect } from "react";
 
 import {
   EntitySeriesWrapper,
   SeriesApiResponse,
 } from "../../../shared/stat_types";
 import { stringifyFn } from "../../../utils/axios";
-import { ChartDataType, ChartStoreAction } from "../chart_store";
+import { ChartDataType, ChartStore, ChartStoreAction } from "../chart_store";
+import { useCalculateRatio } from "../compute_hook";
+import { Context } from "../context";
+import { useDenomStatReady } from "../ready_hook";
 
-export function fetchDenomStat(
-  parentEntity: string,
-  childType: string,
-  statVar: string,
+export function useFetchDenomStat(
+  chartStore: ChartStore,
   dispatch: Dispatch<ChartStoreAction>
 ): void {
-  const action: ChartStoreAction = {
-    type: ChartDataType.DENOM_STAT,
-    error: null,
-    context: {
-      placeInfo: {
-        enclosingPlace: {
-          dcid: parentEntity,
-          name: "",
+  const { placeInfo, statVar } = useContext(Context);
+  const denomStatReady = useDenomStatReady(chartStore);
```

Comment:
same questions as for useBreadcrumbDenomStat about why we need to check calculateRatio and why we need to check denomStatReady

Reply:
ditto